packages/calcite-components/src/components/dialog/dialog.tsx

```diff
@@ -380,7 +396,9 @@ export class Dialog
 
   onOpen(): void {
     this.calciteDialogOpen.emit();
-    activateFocusTrap(this);
+    if (this.modal || (this.modal === false && this.focusTrapDisabled === false)) {
```

Comment:
maybe move this inside `handleFocusTrapDisabled` and call that private method here.